### PR TITLE
Embark fixes

### PR DESCRIPTION
**Pull request by sampettersson**
See commits for what 😇

### CODE REVIEW

File: Projects/Embark/Sources/Viewables/Actions/TextActionSet.swift [hunk 1]
```diff
@@ -37,19 +37,20 @@ extension TextActionSet: Viewable {
...
             }
 
+            let endIndex = (data.textActionSetData?.textActions.endIndex ?? 1)
+            let isLastAction: Bool = index == endIndex - 1
+
             let input = EmbarkInput(
                 placeholder: textAction.data?.placeholder ?? "",
                 keyboardType: masking?.keyboardType,
                 textContentType: masking?.textContentType,
+                returnKeyType: isLastAction ? .done : .next,
```
Comment (tarikstafford) [inline Projects/Embark/Sources/Viewables/Actions/TextActionSet.swift:49]: nice add

File: Projects/Embark/Sources/Viewables/Actions/Masking.swift [hunk 2]
```diff
@@ -14,8 +14,25 @@ enum MaskType: String {
...
     let type: MaskType
 
-    func isValid(text _: String) -> String {
-        ""
+    func isValid(text: String) -> Bool {
+        switch type {
+        case .personalNumber:
+            return text.count > 10
+        case .birthDate, .birthDateReverse:
+            return text.count == 10
```
Comment (tarikstafford) [inline Projects/Embark/Sources/Viewables/Actions/Masking.swift:22]: if the formatting is slightly different from be won't this fail? I saw on the Moving Flow - birthdate was formatted without dashes. Don't we get the max min from BE for validation?

**Review by tarikstafford — APPROVED** [review]
Looks good, just concerned about birthdate validation!

File: Projects/Embark/Sources/Viewables/Actions/Masking.swift (hunk 2 above)
Comment (sampettersson) [reply Projects/Embark/Sources/Viewables/Actions/Masking.swift:22]: The function here makes sure the formatting is always correct: https://github.com/HedvigInsurance/ugglan/blob/e967efe69cb87d508897852681ae6172ed28c75a/Projects/Embark/Sources/Viewables/Actions/Masking.swift#L176

I am guessing in the case of moving flow a mask hasn't been used/set at all in angel, but in that case this function wouldn't be called at all 😄 We should also implement min/max though.

**Review by tarikstafford — APPROVED** [review]

**Pull request merged**